### PR TITLE
Added a 'prepare' step as part of creation of VM (see also #963)

### PR DESCRIPTION
**Pull request by mzsilak**
This is an implementation of feature request #963

### CODE REVIEW

**Comment by retr0h** [comment]
Hi @mzsilak I'll check this out.

One thing I was thinking.  Maybe use the terminology customize vs prepare, but we can talk about that later.  I'll look over the code tho.

**Comment by kireledan** [comment]
This looks pretty interesting. I'm definitely a fan of this.

File: molecule/cookiecutter/scenario/driver/delegated/{{cookiecutter.molecule_directory}}/{{cookiecutter.scenario_name}}/prepare.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+{% raw -%}
+- name: Prepare
+  hosts: all
+  gather_facts: False
+  tasks:
+    - name: Install python for Ansible
+      raw: test -e /usr/bin/python || (apt -y update && apt install -y python-minimal)
```
Comment (kireledan) [inline molecule/cookiecutter/scenario/driver/delegated/{{cookiecutter.molecule_directory}}/{{cookiecutter.scenario_name}}/prepare.yml]: Would it be better to use the 'package' module for the 'default' prepare step? That way non apt package managers don't get errors on first run.

**Comment by mzsilak** [comment]
Hi @retr0h, apart from the terminology I also thought about doing the _prepare_ similarly to _side_effects_. Meaning only execute it when it is set. The term _customize_ would fit to this.
I personally opted for the current way, because the _install python_ task is mandatory anyway.

Hi @kireledan, unless I did a copy and paste mistake, the task was already as it is and I just moved it from playbook.yml to prepare.yml. But I agree, the package module would be better.
I could do it once this merge gets accepted or declined.

**Comment by retr0h** [comment]
I personally like the terminology customize.  I'll review this soon enough.

**Comment by retr0h** [comment]
Closing this one, reimplemented in #971.

**Pull request closed without merging**